api/jobset/v1alpha2/jobset_webhook.go

```diff
@@ -101,6 +101,12 @@ func (js *JobSet) ValidateCreate() (admission.Warnings, error) {
 		if int64(parallelism)*int64(rjob.Replicas) > math.MaxInt32 {
 			allErrs = append(allErrs, fmt.Errorf("the product of replicas and parallelism must not exceed %d for replicatedJob '%s'", math.MaxInt32, rjob.Name))
 		}
+		// Check that the generated job names for this replicated job will be DNS 1035 compliant.
+		// Use job index 0 as the test example.
+		testJobName := fmt.Sprintf("%s-%s-%d", js.Name, rjob.Name, 0)
```

Comment:
can we have a shared func between the webhook and the controller for name generation just so when changes are made, they get reflected in both places.

Reply:
Ok in order to avoid an import/dependency cycle, I had to create a shared util function. However, it didn't seem to fit into any of our existing util pacakges (util/test, util/cert, util/collections) so I made a new one: `pkg/util/names`. I tried to add `GenSubdomain()` into it as well, but it results in an import cycle as well, and unlike `GenJobName` there's no easy fix, so I left it out.

Let me know if you have any thoughts on this.